src/display/api.js

```diff
@@ -151,7 +151,8 @@ if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("PRODUCTION")) {
  *   NOTE: If TypedArrays are used they will generally be transferred to the
  *   worker-thread. This will help reduce main-thread memory usage, however
  *   it will take ownership of the TypedArrays.
- * @property {Object} [httpHeaders] - Basic authentication headers.
+ * @property {Record<string, string>} [httpHeaders] - Basic authentication
+ *   headers.
```

Comment:
Please note that the primary use-case for the JSDoc-comments is to generate https://mozilla.github.io/pdf.js/api/draft/index.html, and consequently we can't have any TypeScript-specific types here; please just revert this change.